packages/payment-detection/src/any/any-to-erc20-proxy-info-retriever.ts

```diff
@@ -101,63 +116,69 @@ export default class ProxyERC20InfoRetriever
     // Get the fee proxy contract event logs
     const feeLogs = await this.provider.getLogs(feeFilter);
 
+    const logToObject = <T>({ args, eventFragment }: LogDescription): T => {
+      return args.reduce((prev, current, i) => {
+        prev[eventFragment.inputs[i].name] = current;
+        return prev;
+      }, {});
+    };
```

Comment:
@vrolland I introduced this utility to avoid `args[0]` and have a better experience, otherwise it's impossible to read!